graphistry/umap_utils.py

```diff
@@ -160,31 +161,28 @@ def umap_lazy_init(
         negative_sample_rate=5,
         n_components: int = 2,
         metric: str = "euclidean",
-        has_umap: str = "True",
-        engine: str = "auto", 
+        engine: str = "auto" #|
     ):
-
-        # FIXME remove as set_new_kwargs will always replace?
         engine_resolved = resolve_umap_engine(engine)
+        # FIXME remove as set_new_kwargs will always replace?
         if engine_resolved == "umap_learn":
             _, _, umap_engine = lazy_umap_import_has_dependancy()
         elif engine_resolved =='cuml':
             _, _, umap_engine = lazy_cuml_import_has_dependancy()
         else:
             raise ValueError("No umap engine, ensure 'auto', 'umap', or 'cuml', and the library is installed")
-        
-        if has_umap and not self.umap_initialized: 
-            umap_kwargs = dict(
-                n_components=n_components,
-                # metric=metric,
-                n_neighbors=n_neighbors,
-                min_dist=min_dist,
-                spread=spread,
-                local_connectivity=local_connectivity,
-                repulsion_strength=repulsion_strength,
-                negative_sample_rate=negative_sample_rate,
-                # engine=engine,
-            )
+
+        if not self.umap_initialized: 
+            umap_kwargs = dict({
+                'n_components':n_components,
+                **({'metric':metric} if engine_resolved=="umap_learn" else {}),
```

Comment:
cool

Reply:
Thinking more: maybe what's needed here is throwing an exn if a non-cuml-allowed val? or would cuml do for us, and we can always put in?